### PR TITLE
feat(onyx-1218): add Collections section to new home view schema

### PR DESCRIPTION
**Pull request by nickskalkin**
Example request:

```graphql
query {
  homeView {
    sectionsConnection(first: 4) {
      edges {
        node {
          ... on MarketingCollectionsRailHomeViewSection {
            component {
              title
            }
            
            marketingCollectionsConnection(first: 1) {
              edges {
                node {
                  title
                }
              }
            }
          }
        }
      }
    }
    
    section(id: "home-view-section-marketing-collections") {
      ... on MarketingCollectionsRailHomeViewSection {
        component {
          title
        }
      }
    }
  }
}
```

Response:

```json
{
  "data": {
    "homeView": {
      "sectionsConnection": {
        "edges": [
          {
            "node": {}
          },
          {
            "node": {}
          },
          {
            "node": {
              "component": {
                "title": "Collections"
              },
              "marketingCollectionsConnection": {
                "edges": [
                  {
                    "node": {
                      "title": "Trending Now"
                    }
                  }
                ]
              }
            }
          },
          {
            "node": {}
          }
        ]
      },
      "section": {
        "component": {
          "title": "Collections"
        }
      }
    }
  }
}
```

### CODE REVIEW

File: _schemaV2.graphql [hunk 1]
```diff
@@ -1509,7 +1509,7 @@ type ArticlesRailHomeViewSection implements GenericHomeViewSection & Node {
     before: String
     first: Int
     last: Int
-  ): ArticleConnection!
+  ): ArticleConnection
```
Comment (nickskalkin) [inline _schemaV2.graphql:1512]: I made ArticleConnection and FairConnection nullable to align with other connections

File: src/schema/v2/homeView/marketingCollectionsResolver.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { GraphQLFieldResolver } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { connectionFromArray } from "graphql-relay"
+import { HomePageMarketingCollectionsModuleType } from "../home/home_page_marketing_collections_module"
+
+export const MarketingCollectionsResolver: GraphQLFieldResolver<
+  any,
+  ResolverContext
+> = async (parent, args, context, info) => {
+  const resolver = HomePageMarketingCollectionsModuleType.getFields().results
```
Comment (MounirDhahri) [inline src/schema/v2/homeView/marketingCollectionsResolver.ts:10]: didn't see this pattern before! 👏

Comment (MounirDhahri) [inline src/schema/v2/homeView/marketingCollectionsResolver.ts:10]:
```suggestion
  const { results } = HomePageMarketingCollectionsModuleType.getFields()
```

**Review by MounirDhahri — APPROVED** [review]
nice one 💯

**Pull request merged**